### ISSUE TITLE
SqlVar.cpp: drop calloc/free for numeric variables

### ISSUE THREAD

**Issue by GitMensch** [mridoni/gix#35]
Seen when needed to fix an upstream bug which is solved here already:
Currently the `real_data` and final data are allocated and freed on each access. This just does not make _any_ sense.

Please adjust this to have either a single static area for numeric "real data" or in each used part.
For anything that isn't float the `COB_MAX_DIGITS` seem enough (38 -> `numeric_real_data[38], similar for packed ((define + 1) / 2).
Float data would be larger.

**Comment by mridoni** [comment]
Yes, this surely needs to be optimized. Even so, the slow down, measured  in real-world applications, is not as big as it might appear.

**Comment by GitMensch** [comment]
Totally - the DB "behind" is slower. With that optimization the code will get smaller, more clear and has less options to go wrong - those are the more relevant parts.

**Comment by GitMensch** [comment]
Also: they area already removed from the "set DB to COBOL" code (most reasonable with directly operating on the program's buffer), so only the "unpacking" code would benefit from this change.
